dev/com.ibm.ws.microprofile.reactive.streams.operators/test/src/com/ibm/ws/microprofile/reactive/streams/test/CollectTest.java

```diff
@@ -0,0 +1,135 @@
+/*******************************************************************************
+ * Copyright (c) 2018 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * The initial set of unit test material was heavily derived from
+ * tests at https://github.com/eclipse/microprofile-reactive
+ * by James Roper.
+ ******************************************************************************/
+
+package com.ibm.ws.microprofile.reactive.streams.test;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+import java.util.stream.Collector;
+import java.util.stream.Collectors;
+
+import org.eclipse.microprofile.reactive.streams.ReactiveStreams;
+import org.eclipse.microprofile.reactive.streams.SubscriberBuilder;
+import org.junit.Test;
+
+public class CollectTest extends WASReactiveUT {
+
+    @Test
+    public void toListStageShouldReturnAList() {
+        assertEquals(await(ReactiveStreams.of(1, 2, 3).toList().run(getEngine())),
+                     Arrays.asList(1, 2, 3));
+    }
+
+    @Test
+    public void toListStageShouldReturnEmpty() {
+        assertEquals(await(ReactiveStreams.of().toList().run(getEngine())),
+                     Collections.emptyList());
+    }
+
+    @Test
+    public void collectShouldAccumulateResult() {
+        assertEquals(await(ReactiveStreams.of(1, 2, 3).collect(
+                                                               () -> new AtomicInteger(0),
+                                                               AtomicInteger::addAndGet).run(getEngine())).get(),
```

Comment:
We can probably change the formatter settings so that it doesn't butcher these methods.

Reply:
I have added a project specific formatting rules file and started improving the rules.